### PR TITLE
Edit UG

### PR DESCRIPTION
**Pull request by zannloo**
Edit UG according to feedback from PE-D

- [x] #258 - Able to add deadline in the past
- [x] #257 
- [x] #256  - Need clear description to clarify the difference of group task and individual task. And also mention allStu is create individual task for all students, allGrp is create group task for all groups. 
- [x] #255 - Add more screenshot into the UG
- [ ] #253 - Not sure what this means 
- [x] #249 
- [x] #248 
- [x] #246 
- [x] #245 
- [x] #242 
- [x] #240 
- [x] #239 
- [x] #236 
- [x] #235 - consider putting in future enhancement
- [x] #234 
- [x] #231 
- [x] #227 
- [x] #225 
- [x] #223 
- [ ] #211 - Not adding this round

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -162,25 +164,33 @@ ProfBook is organized hierarchically with the following structure:
...
 - `--name` / `-n` followed by name of student or group specified by command.
 - `--email` / `-e` followed by email of student. 
 - `--phone` / `-p` followed by phone number of student.
 - `--address` / `-a` followed by address of student.
+- `--id` / `-i` followed by a `STUDENT_ID` or `GROUP_ID` as specified by the command.
 - `--desc` followed by description of task specified by command.
 - `--datetime` / `-dt` followed by due date of a deadline task in the `yyyy-MM-dd HH:mm` format.  
 - `--all` / `-al` followed by either: 
-  - `allStu` to add a task to all students.
+  - `allStu` which adds **individual** tasks to all students under within the group directory.
```
Comment (Nid21cs) [inline docs/UserGuide.md:171]: Should just be under or within not both?

**Review by Nid21cs — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md [hunk 2]
```diff
@@ -142,7 +143,8 @@ ProfBook is organized hierarchically with the following structure:
   **Required format for `StudentId`:**
   - Must be in the format `XXXXA` where `XXXX` is replaceable with any 4 digit number and `A` is replaceable with any
   capitalised alphabet.
-    - This format accurately reflects the last 5 place of an NUS student's matriculation number while maintaining 
+    - - **Note:** StudentId here corresponds to the last 5 place of an NUS student's matriculation number and is not
```
[the remark this answers is not in the document]
Comment (zannloo) [reply docs/UserGuide.md]: However, the last 5 is a mix of digits and alphabet. But good suggestion that suggests that readers may have confusion when reading this part. Will think further on better alternatives!

File: docs/UserGuide.md [hunk 3]
```diff
@@ -549,9 +740,9 @@ Moves student from one group to another group.
 #### Output if command fails
 
 - Displays message indicating either:
-  - [Invalid command format](#mistake-1-invalid-command-format)
+  - [Invalid command format.](#mistake-1-invalid-command-format)
   - [No such student to move.](#mistake-4-no-such-student-to-move)
-  - Invalid destination path.
+  - [Invalid destination path.](#mistake-2--invalid-path)
```
Comment (NgChunMan) [inline docs/UserGuide.md:745]: This link is invalid. Nothing happens after clicking on it. Perhaps the link contains a redundant '-', which causes it to be invalid.

**Review by NgChunMan — COMMENTED** [review]
LGTM. Just a few minor changes need to be made.

**Pull request merged**